pkg/kube/inject/inject.go

```diff
@@ -649,24 +655,32 @@ func IntoObject(sidecarTemplate Templates, valuesConfig string, revision string,
 		ObjectMeta: *metadata,
 		Spec:       *podSpec,
 	}
-	if !injectRequired(ignoredNamespaces, &Config{Policy: InjectionPolicyEnabled}, &pod.Spec, pod.ObjectMeta) {
-		warningHandler(fmt.Sprintf("===> Skipping injection because %q has sidecar injection disabled\n", name))
-		return out, nil
+
+	var patchBytes []byte
+	var err error
+	if injector != nil {
+		patchBytes, err = injector.Inject(pod)
```

Comment:
by ignoring `err` here we make it very hard to understand what is going wrong here

Reply:
returned err here since  we have set injector as nil when we are falling back to use configmap.